samples/app-anonymous-users/csharp/ClientApp/src/components/shareview.jsx

```diff
@@ -84,11 +112,195 @@ const ShareView = () => {
+                    },
+                    cache: 'default'
+                })
+                    .then((response) => {
+                        if (response.ok) {
+                            return response.text();
+                        } else {
+                            reject(response.error);
+                        }
+                    })
+                    .then((responseJson) => {
+                        if (responseJson === "") {
+                            IsSetVisibleConsent(true);
+                            ssoSetAuthenticationBtn(false);
+                        } else {
+                            let userDetails = JSON.parse(responseJson);
+                            let userName = userDetails.user.displayName;
+                            setUserName("Welcome: " + userName);
+                            ssoSetAuthenticationBtn(false);
+                            setEnableVoteDiv(true);
+                        }
+                    });
+            });
+        });
+    }
+
+    // Request consent on implicit grant error.
+    const requestConsent = () => {
+        getToken()
+            .then(data => {
+                IsSetVisibleConsent(false);
+                getClientSideToken()
+                    .then((clientSideToken) => {
+                        return getServerSideToken(clientSideToken);
+                    });
+            });
+    }
+    // Get token for multi tenant.
+    const getToken = () => {
+        return new Promise((resolve, reject) => {
+            microsoftTeams.authentication.authenticate({
+                url: window.location.origin + "/auth-start",
+                width: 600,
+                height: 535
+            })
+                .then((result) => {
+                    resolve(result);
+                })
+                .catch((reason) => {
+                    reject(reason);
+                });
+        });
+    }
+
     return (
         <div className="timerCount">
-            <Button appearance="primary" onClick={submitVote} disabled={disabled} >Submit Vote</Button>
-            <h1>Anonymous users voted: {aShowCount}</h1>
-            <h1>Users voted: {uShowCount}</h1>
+            {IsCheckLoggeduser
+                ? <div className="btnlogin">
+                    {ssoAuthenticationBtn &&
+                        <Button appearance="primary" onClick={ssoAuthentication}>Sign-In</Button>
+
+                    }
+                    <Text size={500} weight="semibold">{userName}</Text>
+                    {IsVisibleConsent &&
+                        <>
+                            <div id="divError">Please click on consent button</div>
+                            <Button appearance="primary" onClick={requestConsent}>Consent</Button>
+                        </>
+                    }
+                </div>
+                : <div className="btnlogin">
+                    {disabledfacebookBtn &&
```

Comment:
Can re rename `disabledfacebookBtn` so that when it's true the facebook login button is disabled, instead of now, where when `disabledfacebookBtn` is true the button is shown

Reply:
Updated your feedback.